docs/java/features/odata/type-safe-client-odata-v4.mdx

```diff
@@ -211,45 +211,35 @@ The described API is subject to change
 #### Execute Batch Request
 The `service` object offers the method `batch` which allows access to the batch-related methods that help build the batch request.
 
-Multiple single requests (like `createBusinessPartnerAddress`) can be wrapped into so-called change sets.
-Each change set is opened with `beginChangeSet` and closed with `endChangeSet`. You can wrap multiple change sets into one batch request.
+Multiple single requests (like `createBusinessPartner`) can be wrapped into so-called change sets.
+A change set can be added by `addChangeset` and the corresponding create, update or delete operations can be passed as partmenters into it. You can wrap multiple change sets into one batch request.
+Similarly, retrieve operations can be added to the batch request by using `addReadOperations`.
 
-Use `executeRequest` to issue the batch request to the remote system. We receive an instance of `BatchResponse` as result object.
+Use `execute` to issue the batch request to the remote system. We receive an instance of `BatchResponse` as result object.
 
 ```java
 final BatchResponse result =
     service
         .batch()
-        .beginChangeSet()
-        .createBusinessPartnerAddress(address1)
-        .createBusinessPartnerAddress(address2)
-        .endChangeSet()
-        .executeRequest(destination);
+        .addReadOperations(businessPartners, businessPartnerByKey)
+        .addChangeset(createBusinessPartner1(), createBusinessPartner2(), deleteBusinessPartner())
```

Comment:
For clarity consider to rename the builders to something like `getAllRequest, createRequest`. Otherwise from the code it's unclear that these are request builder objects and not actual business partners.

Reply:
Done